linkedList.js

```diff
@@ -4,5 +4,81 @@
  * @return {Object} an object exposing methods to be used to manipulate a linked list
  */
 function linkedListGenerator(){
+    let head = null;
+    let tail = null;
+    return {
+        getHead: function(){
+            return head;
+        },
+        getTail: function(){
+            return tail;
+        },
+        add: function(value){
+            let newNode = new Object;
+            newNode.value = value;
+            newNode.next = null;
+            if (!head){
+                head = newNode;
+                tail = newNode;
+            } else {
+                tail.next = newNode;
+                tail = newNode;
+            }
+            return newNode;
+        },
+        get: function(index){
+            let current = head;
 
+            let i = 0;
+            while (current !== null) {
+                if (i === index) {
+                    return current;
+                } else { 
+                    current = current.next;
+                    i++;
+                }
+            }
+            return false;
+         
+        },
+        remove: function(index){
+            let current = this.get(index);
+            let before, after;
+            if (this.get(index) === false){
+                return false;
+            }
+            if (index > 0){
+                before = this.get(index - 1);
+            } else {
+                before = false;
+                head = current.next;
+            }
+            if (this.get(index + 1)){
```

Comment:
So having these two if/else statements one right after the other, could run into a situation where the head gets removed, and tail gets set to false. It also a bit hard to read and understand the logic. Maybe try to handle it all in one continue if/else if/else if/else.